src/main/java/ch/njol/skript/doc/JSONGenerator.java

```diff
@@ -0,0 +1,247 @@
+import com.google.gson.JsonObject;
+import com.google.gson.JsonPrimitive;
+import org.eclipse.jdt.annotation.Nullable;
+import org.skriptlang.skript.lang.structure.Structure;
+import org.skriptlang.skript.lang.structure.StructureInfo;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Iterator;
+import java.util.Objects;
+import java.util.stream.Stream;
+
+/**
+ * Generates JSON docs
+ */
+public class JSONGenerator extends DocumentationGenerator {
+
+	public JSONGenerator(File templateDir, File outputDir) {
+		super(templateDir, outputDir);
+	}
+
+	/**
+	 * Coverts a String array to a JsonArray
+	 * @param strings the String array to convert
+	 * @return the JsonArray containing the Strings
+	 */
+	private static @Nullable JsonArray convertToJsonArray(String @Nullable[] strings) {
+		if (strings == null)
+			return null;
+		JsonArray jsonArray = new JsonArray();
+		for (String string : strings)
+			jsonArray.add(new JsonPrimitive(string));
+		return jsonArray;
+	}
+
+	/**
+	 * Generates the documentation JsonObject for an element that is annotated with documentation
+	 * annotations (e.g. effects, conditions, etc.)
+	 * @param syntaxInfo the syntax info element to generate the documentation object of
+	 * @return the JsonObject representing the documentation of the provided syntax element
+	 */
+	private @Nullable JsonObject generatedAnnotatedElement(SyntaxElementInfo<?> syntaxInfo) {
+		Class<?> syntaxClass = syntaxInfo.getElementClass();
+		Name nameAnnotation = syntaxClass.getAnnotation(Name.class);
+		if (nameAnnotation == null || syntaxClass.getAnnotation(NoDoc.class) != null)
+			return null;
+		JsonObject syntaxJsonObject = new JsonObject();
+		syntaxJsonObject.addProperty("id", DocumentationIdProvider.getId(syntaxInfo));
+		syntaxJsonObject.addProperty("name", nameAnnotation.value());
+
+		Since sinceAnnotation = syntaxClass.getAnnotation(Since.class);
+		syntaxJsonObject.addProperty("since", sinceAnnotation.value());
+
+		Description descriptionAnnotation = syntaxClass.getAnnotation(Description.class);
+		if (descriptionAnnotation != null)
+			syntaxJsonObject.add("description", convertToJsonArray(descriptionAnnotation.value()));
+		else
+			syntaxJsonObject.add("description", new JsonArray());
+
+		Examples examplesAnnotation = syntaxClass.getAnnotation(Examples.class);
+		if (examplesAnnotation != null) {
+			syntaxJsonObject.add("examples", convertToJsonArray(examplesAnnotation.value()));
+		} else {
+			syntaxJsonObject.add("examples", new JsonArray());
+		}
+
+
+		syntaxJsonObject.add("patterns", convertToJsonArray(syntaxInfo.getPatterns()));
+		return syntaxJsonObject;
+	}
+
+	/**
+	 * Generates the documentation JsonObject for an event
+	 * @param eventInfo the event to generate the documentation object for
+	 * @return a documentation JsonObject for the event
+	 */
+	private JsonObject generateEventElement(SkriptEventInfo<?> eventInfo) {
```

Comment:
Shouldn't this check if the event has no doc?

Reply:
i don't think events really have that system. there's a NO_DOC array but its not like a flag, it's just an empty string array you are meant to provide as the description if there is no description. if it's null, that also gets handled by `convertToJsonArray`